### ISSUE TITLE
Package nvidia-driver-390

### ISSUE THREAD

**Issue by PanderMusubi** [pop-os/nvidia-graphics-drivers-470#22]
I have noticed that the package nvidia-driver-390 is supported by Ubuntu Jammy 22.04 LTS, Lunar 23.04, but not in Mantic 23.10 and Noble 24.04 LTS.

Will the package nvidia-driver-390 be available in the next release of Pop!_OS?

**Comment by jackpot51** [comment]
We package 470 because it is the last version that supports the 700 series and earlier GPUs. Is there a set of GPUs that are not supported by 470?

**Comment by PanderMusubi** [comment]
Yes, specifically GeForce GTX 560 Ti for me. Besides my System76 laptop, I run Pop!_OS successfully on all my desktops of other brands, some with older hardware.

When the 390 driver package is shipped in Pop!_OS, it will work out of the box with [519 GPUs](https://download.nvidia.com/XFree86/Linux-x86_64/550.76/README/supportedchips.html), mainly of type Geforce. Installing the NVIDIA driver manually is not straight forward and will create some package dependency issues.

Supporting this will draw in Ubuntu users to Pop!_OS that cannot upgrade otherwise and it will also reduce e-waste as users don't have to get newer GPUs, which in turn can also require heavier power units in desktops.

**Comment by PanderMusubi** [comment]
@jackpot51 did you and others had a change to discuss this? To me, it seems like a good opportunity to get more users.

**Comment by PanderMusubi** [comment]
See also:
- https://launchpad.net/ubuntu/+source/nvidia-graphics-drivers-390
- https://bugs.launchpad.net/ubuntu/+source/nvidia-graphics-drivers-390/+bug/2064953
- https://www.phoronix.com/forums/forum/linux-graphics-x-org-drivers/nvidia-linux/1343469-nvidia-proprietary-driver-causes-last-minute-headache-for-ubuntu-20-04-5-lts#post1343496

**Comment by sohamthakurdesai** [comment]
@PanderMusubi thanks for raising this issue. I came across this discussion while setting up my old laptop for gaming. I have 12-13 year old Dell XPS L501X. It has got Nvidia GeForce 420m GPU. Installing nvidia-driver-390 on it while using Pop!OS has been quite cumbersome. As you mentioned I am facing many package dependencies and DKMS issues. So far I am not able to achieve the goal.

I am opting for Pop!OS because its Nvidia hardware support is good and also because it is maintained by System76. I would definitely appreciate if out-of-the-box support is provided for nvidia-driver-390.

Looking forward to your team's reply on it! @jackpot51 :) 

@PanderMusubi have you got any work around?

**Comment by wildrun0** [comment]
Still no 390 driver...

**Comment by Oredukri** [comment]
Same issue here, luckily the integrated one will keep me from scrapping the ol laptop.   @jackpot51 The geforce series 400-635M are not covered by 470.

**Comment by PanderMusubi** [comment]
I the spirit of https://netflix.com/title/81554996 keeping support for older GPUs is reducing a lot of e-waste.

Anybody using https://nouveau.freedesktop.org/CodeNames.html ?

**Comment by Oredukri** [comment]
@PanderMusubi I tried nouveau on my 630m but the performance was way worse than the one from my integrated graphics so I rolled back to that one.
 
As always, your experience might vary.

**Comment by PanderMusubi** [comment]
Very unfortunate. What would be another community to find traction for supporting this driver? Perhaps in the kernel module community? Or some funding could be found for this as it will prevent a lot of hardware being wasted.

**Comment by mmstick** [comment]
@PanderMusubi It's up to NVIDIA to make their drivers compatible with the recent stable kernels. The drivers aren't open source.

**Comment by T100D** [comment]
@PanderMusubi NVIDIA is moving to open source so things will change

https://developer.nvidia.com/blog/nvidia-transitions-fully-towards-open-source-gpu-kernel-modules/

**Comment by PanderMusubi** [comment]
> @PanderMusubi It's up to NVIDIA to make their drivers compatible with the recent stable kernels. The drivers aren't open source.

Isn't there a way to add a layer in between so newer kernels can still use older closed drivers?

**Comment by PanderMusubi** [comment]
> @PanderMusubi NVIDIA is moving to open source so things will change
> 
> https://developer.nvidia.com/blog/nvidia-transitions-fully-towards-open-source-gpu-kernel-modules/

Thanks. Unfortunately, this only concerns the newer 555 driver, not? See also https://www.phoronix.com/news/NVIDIA-Transitions-OSS-KMD and the comments.

**Comment by mmstick** [comment]
@PanderMusubi No, that layer would be for NVIDIA to develop open source kernel modules, which is being developed only for currently-supported architectures.

**Comment by Oredukri** [comment]
I doubt we'll see steps taken by NVIDIA regarding hardware they deem to be at the EOL, with any luck some of the community updates to 390 for modern kernels will trickle down to Pop

**Comment by h-n-a** [comment]
Stuck in this same boat with a Thinkpad W520 and switchable Nvidia Quadro 1000M. i've opted to install Pop_os for the driver support as well, but i guess we are stuck with nouveau driver which is so bad.
Even a simple game like Terraria from 2011, will not run smoothly until you disable bunch of stuff in the game.
The GPU is capable of running games like "Rainbow Six: Vegas 2" and several other titles but because we don't have hardware acceleration with the proprietary driver, its not running any games for me from my library on steam.

**Comment by PanderMusubi** [comment]
Marge we should create an issue at NVIDIA and gather support (+1s) over there.

**Comment by h-n-a** [comment]
According to Nvidia and someone i read on mint linux, Nvidia no longer want to support 390.157 on the newer kernels.

![Image](https://github.com/user-attachments/assets/92da4cd6-3cc0-4d02-886b-f7960491080d)

https://forums.developer.nvidia.com/t/390-157-driver-no-longer-works-with-kernel-6-5/283941

![Image](https://github.com/user-attachments/assets/c1f84330-dc63-4a40-a316-113bdaae6194)

https://forums.linuxmint.com/viewtopic.php?t=427160

I also tried following:

https://www.if-not-true-then-false.com/2021/debian-ubuntu-linux-mint-nvidia-guide/

but ended up breaking Wayland/X11

**Comment by PanderMusubi** [comment]
I guess, now two years after the official support has ended, there is a slim to none chance they will support kernel 6. Don't think any form of petition will convince NVIDIA to change its mind and prevent a lot of e-waste. But whoever wants, feel free to ask them.

Nouveau is now all you can use. How well that works depends heavily on the support of Nouveau for your card and what you want to do with it.

For me personally, this issue can be closed and will move to newer GPU cards Thanks all for looking into this.

**Comment by h-n-a** [comment]
Like i said, Nouveau driver is crap. barely does anything to hold the water.
I think our only saving grace would be to install an eGPU dock and salvage whats left of the laptop using that.
Never seen anyone try that in linux like they do in windows. 

And about the e-Waste topic, i agree with you. 
All these self righteous companies parading around "climate change" propaganda but don't want to hold themselves to that standard. Just bunch of hypocrites.
Look at all the android phones that have been tossed away because they no longer receive updates.

**Comment by dazhuklev** [comment]
Unfortunately, that was the reason for me to go back to Windows after so, so many years running Linux. But to be fair, it's not only NVIDIA's fault but also the Linux kernel developers because, e.g. the driver on Windows is from 2016 and still works. Even after all the Windows updates. There are no driver updates since 2016 for my hardware which is NVIDIA Quadro K2000M on Lenovo W530 but the driver still works. And it works pretty well to be honest, I can finally utilize my hardware which I haven't been able to for a while and it was very annoying. 

Again, my post was mostly to be fair on who to "blame" for the lack of support for old hardware on newer OS releases. 

For now, I'm obviously stuck with Windows but the good thing is that I can also continue using Ubuntu through WSL2 which is great, so I kind of get the best of both worlds on my old hardware. :)

**Comment by bluecenterlight** [comment]
TY for this thread! I've been wanting to return to Linux and heard of POP OS's driver compatibility. You saved me hours of tinkering work and frustration since I know I'll have the same problems trying to get the 390 driver to work.  Staying with Win 10 LTSC iot since the 2016 nvidia driver still works well.

**Comment by PanderMusubi** [comment]
Latest driver is from Tue Nov 22, 2022. C'est la vie, despite all the other sustainability they do https://www.nvidia.com/en-us/foundation/archives/

**Comment by EmmanuelChanel** [comment]
Hello! I don't know Pop!OS well. But if it's based on Ubuntu 24.04 LTS, you can install nvidia driver 390 by installing these PPA.
https://launchpad.net/~kelebek333/+archive/ubuntu/nvidia-legacy
https://launchpad.net/~kelebek333/+archive/ubuntu/build
I could "solve" my driver problem on Linux Mint 22.
See also: https://forums.linuxmint.com/viewtopic.php?t=425507

**Comment by PanderMusubi** [comment]
Thanks for the links. If you can motivate the maintainer to also support 26.04 LTS, that would be great.

**Comment by h-n-a** [comment]
@EmmanuelChanel  can you provide more details? your response doesn't say much. 
i did run into that driver. but the installation steps aren't clear since it fails.
is there anything special you had to do?
What kernel are you using?

**Comment by EmmanuelChanel** [comment]
> [@EmmanuelChanel](https://github.com/EmmanuelChanel) can you provide more details? your response doesn't say much. i did run into that driver. but the installation steps aren't clear since it fails. is there anything special you had to do? What kernel are you using?

Well, my kernel is like this. I don't really remember the detail since it's about 1 and a quarter years ago.

>$ ls /boot/
>System.map-6.8.0-88-generic  initrd.img-6.8.0-88-generic
>System.map-6.8.0-90-generic  initrd.img-6.8.0-90-generic
>config-6.8.0-88-generic      initrd.img.old
>config-6.8.0-90-generic      vmlinuz
>efi                          vmlinuz-6.8.0-88-generic
>grub                         vmlinuz-6.8.0-90-generic
>initrd.img                   vmlinuz.old

But this helped me, as far as I remember.

>sudo add-apt-repository ppa:kelebek333/build
>sudo apt update
>sudo apt install nvidia-legacy-390xx-driver

Anyway, this would fail since nvidia-leacyt-390xx-driver hasn't been ported yet.

>sudo add-apt-repository ppa:kelebek333/nvidia-legacy
>sudo apt update

**Comment by h-n-a** [comment]
ah, as i suspected, we have to downgrade our kernels from 6.17.x to 6.8.

**Comment by h-n-a** [comment]
I spent 6 hours beating my head trying to downgrade the kernel on popos, it just won't work. i can't get it to point to the 6.8 kernel. "Kernelstub" command is running into an issue  ¯\\_(ツ)_/¯
So Ended up installing Linux Mint 22.1 which comes with Kernel 6.8

Be ware that "sudo apt install nvidia-legacy-390xx-driver" will not work, it will just error out and say that it couldn't find the file.
No one out there has a clear or correct way to install it so i improvised.

So what i ended up doing:
1-First went into the Thinkpad W520 bios, and changed the display option to discrete(Nvidia)
2-Installed Linux Mint 22.1
3-Installed "Synaptic" package manager from Ubuntu
"sudo apt install synaptic"
4-Added "sudo add-apt-repository ppa:kelebek333/build" in the terminal
5-Launched Synaptic Manager.
6-From the left menu in the Synaptic Manager, select "origin"
7-Click on LP-PPA-Kelebek333-build/noble
8-on the right side pane, right click on any thing that has "390" in it (that's what i did whether its the right way or not) and click on "Mark for installation"
-in Step 8 you might be prompted on some files that it will mark more files automatically as they are dependencies.
9-Click on "apply" from the top left in synaptic package manager

And as you can see below from the terminal, "v:390.157" driver is loaded. but i guess that nouveau has to be blacklisted or purged to prevent conflict(?)

hopefully that helps someone else here.
i don't like the fact that the laptop is stuck with 6.8. im not an expert enough on linux to tell whether there will be a hazard or not, but im guessing that security updates are in the newer kernel releases.

The next thing i need to figure out, how to get the same option that popos has to switch between the intel and nvidia gpus.

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/9ade2d89-2510-4225-bd78-ccae529365f7" />
